src/shifter.c

```diff
@@ -166,13 +166,20 @@ int main(int argc, char **argv) {
         fprintf(stderr,"FAILED permission denied to image\n");
         exit(1);
     }
+    if ( (opts.args==NULL || opts.args[0]==NULL) && imageData.cmd!=NULL && opts.entrypoint==NULL) {
+        opts.args=imageData.cmd;
+    }
 
     /* check if entrypoint is defined and desired */
     if (opts.useEntryPoint == 1) {
-        char *entry = NULL;
+        char **entry = NULL;
 
         if (opts.entrypoint != NULL) {
-            entry = opts.entrypoint;
+            entry = make_char_array(opts.entrypoint);
+            if (entry == NULL) {
+                fprintf(stderr, "Failed to allocate memory for entry\n");
```

Comment:
this error check should be unnecessary in modern dev branch shifter (hopefully all merged).  make_char_array should use one of the shifter override allocation routines that abort()s if memory allocation fails

Reply:
Can we leave this in place for now?

Reply:
we'll have to remember to remove it and fix make_char_array() once the modules branch gets merged